src/commands/fetch/fetch.rs

```diff
@@ -10,14 +10,28 @@ pub async fn fetch(ctx: &client::Context, msg: &Message, mut args: Args) -> Comm
     let db = ctx.get_db().await;
 
     let guild = msg.guild(&ctx).await.context("Failed to load guild")?;
+    let mut desired_field: Option<String> = None;
     let mentioned_user_id = match args.single_quoted::<String>() {
-        Ok(mentioned_user) => disambiguate_user_mention(&ctx, &guild, msg, &mentioned_user)
-            .await?
-            .ok_or(UserErr::MentionedUserNotFound)?,
+        Ok(mentioned_user) => {
+            if find_fetch_key_matching(&mentioned_user).is_some()
+                || IMAGE_KEY == mentioned_user.to_lowercase()
+            {
+                println!("does contain");
+                desired_field = Some(mentioned_user);
+                msg.author.id
+            } else {
+                disambiguate_user_mention(&ctx, &guild, msg, &mentioned_user)
+                    .await?
+                    .ok_or(UserErr::MentionedUserNotFound)?
+            }
+        }
         Err(_) => msg.author.id,
     };
 
-    let desired_field = args.single_quoted::<String>().ok();
+    let desired_field = match desired_field {
+        None => args.single_quoted::<String>().ok(),
+        field => field,
+    };
```

Comment:
These could be combined into one match that returns a tuple, like
```
let (mentioned_user_id, desired_field) = match .....
```
thus cleaning this up a bit

Reply:
all good